src/components/calcite-split-button/calcite-split-button.tsx

```diff
@@ -0,0 +1,145 @@
+import { Component, Element, Event, EventEmitter, h, Host, Prop, Watch } from "@stencil/core";
+import { Scale } from "../../interfaces/common";
+import { getElementDir } from "../../utils/dom";
+
+@Component({
+  tag: "calcite-split-button",
+  styleUrl: "calcite-split-button.scss",
+  shadow: true
+})
+export class CalciteButtonWithDropdown {
+  @Element() el: HTMLElement;
+
+  /** specify the color of the control, defaults to blue */
+  @Prop({ mutable: true, reflect: true }) color: "blue" | "dark" | "light" | "red" = "blue";
+
+  /** select theme (light or dark), defaults to light */
+  @Prop({ mutable: true, reflect: true }) theme: "light" | "dark" = "light";
+
+  /** specify the scale of the control, defaults to m */
+  @Prop({ mutable: true, reflect: true }) scale: "s" | "m" | "l" = "m";
+
+  /** specify the icon used for the dropdown menu, defaults to chevron */
+  @Prop({ mutable: true, reflect: true }) dropdownIconType: "chevron" | "caret" = "chevron";
+
+  /** text for primary action button  */
+  @Prop({ reflect: true }) primaryText: string;
+
+  /** optionally pass an icon to display on the primary button - accepts Calcite UI icon names  */
+  @Prop({ reflect: true }) primaryIcon?: string;
+
+  /** optionally pass an aria-label for the primary button */
+  @Prop({ reflect: true }) primaryLabel?: string;
+
+  /** aria label for overflow button */
+  @Prop({ reflect: true }) dropdownLabel?: string;
+
+  /** optionally add a calcite-loader component to the control,
+    disabling interaction. with the primary button */
+  @Prop({ reflect: true }) loading?: boolean = false;
+
+  /** is the control disabled  */
+  @Prop({ reflect: true }) disabled?: boolean;
+
+  /** fired when the modal begins the open animation */
+  @Event() primaryButtonClicked: EventEmitter;
+
+  @Watch("color")
+  validateColor() {
+    let color = ["blue", "red", "dark", "light"];
+    if (!color.includes(this.color)) this.color = "blue";
+  }
+
+  @Watch("scale")
+  validateScale() {
+    let scale = ["xs", "s", "m", "l", "xl"];
+    if (!scale.includes(this.scale)) this.scale = "m";
+  }
+
+  @Watch("theme")
+  validateTheme() {
+    let theme = ["dark", "light"];
+    if (!theme.includes(this.theme)) this.theme = "light";
+  }
+
+  @Watch("dropdownIconType")
+  validateDropdownIconType() {
+    let dropdownIconType = ["chevron", "caret"];
+    if (!dropdownIconType.includes(this.dropdownIconType)) this.dropdownIconType = "chevron";
+  }
+
+  connectedCallback() {
+    this.validateColor();
+    this.validateScale();
+    this.validateTheme();
+    this.validateDropdownIconType();
+  }
+
+  render() {
+    const dir = getElementDir(this.el);
+    return (
+      <Host dir={dir}>
+        <div>
+          <calcite-button
```

Comment:
@kstinson14 What are your thoughts on having the `calcite-button` and `calcite-dropdown` as slots instead of being created internally and passing all the props? My concern is that as button adds props, this component needs to as well.

Maybe another option is to have just two props `buttonOptions` and `menuOptions` that could be typed as `CalciteButtonProperties`, etc.

Then it could just spread those options on the element like `<calite-button ...buttonOptions>`?

The slot might be more HTML consistent though but you would have to then query for the slotted element and set some of the props like iconPosition, onClick etc.

Reply:
Ah, @macandcheese had a similar question which I answered above.  Basically, I see the main benefit of this component the stylistic combination of the two buttons and the dropdown.

The component lets the user pass in properties once, and have them propagate to all three sub-components.  Colors, scales, etc. are matched without making the user declare them all three times, and without the user needing to worry about inconsistencies between the three (scales for buttons go from `xs` to `xl`, but dropdowns only go `s` to `l` for example).

There's not much internal logic that the component provides, it's mainly this internal alignment of the component visuals.  If we were to slot in the button and dropdown (or use the options properties you describe) and make that alignment something the user had to do themselves, I feel like we'd be losing a lot of the value of the component.

I definitely agree that we should be cautious about how many properties we add to things.  But I don't think that's necessarily something we need to worry about here. I think the control that this component creates is fairly defined.  I don't think it _needs_ to support all of the properties/variations that `calcite-button` has or will receive.  There's several that it doesn't use now, and ones that it makes decisions about internally.  I don't think restricting the aspects of the buttons you can customize here is undesirable.

Reply:
> The component lets the user pass in properties once, and have them propagate to all three sub-components.

👍 

Agreed that this is worth not slotting then.

Reply:
Sounds good, we can always re-evaluate later if a need comes up for more button props to be used here.